src/libcore/ptr/non_null.rs

```diff
@@ -213,6 +221,8 @@ impl<T: ?Sized> From<Unique<T>> for NonNull<T> {
 impl<T: ?Sized> From<&mut T> for NonNull<T> {
     #[inline]
     fn from(reference: &mut T) -> Self {
+        // SAFETY: A mutable reference cannot be null, so the conditions for
+        // new_unchecked() are respected.
         unsafe { NonNull { pointer: reference as *mut T } }
```

Comment:
You refer to `new_unchecked` in the `SAFETY` comment, but it's not what's called here. We should always be using `new_unchecked` to create `NonNull`s, even internally.

Reply:
@ecstatic-morse Do you think I should add it to this PR ? I feel like changing the code is a bit outside of its scope. I'm changing the `SAFETY` comment